bubblewrap.c

```diff
@@ -1327,6 +1333,10 @@ main (int    argc,
   if (!is_privileged)
     opt_unshare_user = TRUE;
 
+  if (opt_unshare_user_try &&
+      stat ("/proc/self/ns/user", &sbuf) == 0)
```

Comment:
Its true that it the checking for userns support on rhel7 will not detect that the clone will not work. However, In practice its not necessarily a problem for RHEL7, because unprivileged userns never works on RHEL7, so any working setup will be using the privileged mode and not running into the issue.

If we were able to detect that userns isn't available (due to RHEL7) and we were not privileged, we can't do anything, not even clone(). So, while its true that --unshare-user-try doesn't work, neither does anything else. such a setup is just broken.

The case that is problematic is when you're privileged, but user ns doesn't work *anyway*. In that case (such as the Arch example) the code here detects at least the case where its completely disabled. One could imagine a different RHEL7 patch that had the proc userns file but disabled userns even for root, and in that case the patch would not help. But is anyone using anything like that?